vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3236,6 +3272,23 @@ public boolean isMultiSort() {
         return Boolean.parseBoolean(multiSort);
     }
 
+    /**
+     * Gets whether multiple column sorting with shift-click is enabled.
+     *
+     * @see #setMultiSort(boolean, boolean)
+     *
+     * @return {@code true} if multi-sort with shift-click is enabled,
+     *         {@code false} otherwise
+     */
+    public boolean isMultiSortShiftClickOnly() {
+        String onShiftClickOnly = getElement()
+                .getAttribute("multi-sort-on-shift-click");
+        if (onShiftClickOnly != null && onShiftClickOnly.length() == 0) {
+            onShiftClickOnly = "true";
+        }
+        return isMultiSort() && Boolean.parseBoolean(onShiftClickOnly);
```

Comment:
Would it be enough to `return getElement().hasAttribute("multi-sort-on-shift-click");` (or rather `return getElement().getProperty("multiSortOnShiftClick", false);`)?

Reply:
Yes and done.

Reply:
One thing to note is that I was copying the `isMultiSort` implementation so maybe that needs to be changed in the future for consistency.